src/pages/home/HomeComponent.js

```diff
@@ -51,29 +57,106 @@ export default ({
+      feedbackElement.hidden = false;
+      feedbackElement.style.color = 'green';
+      setData({ posts: [{ id: postId, ...post }, ...state.posts] });
+      showPosts();
+      setTimeout(() => {
+        feedbackElement.hidden = true;
+      }, 3000);
+    } else {
+      e.target.style.borderColor = 'red';
+      feedbackElement.innerHTML = 'Mensaje es requerido!';
+      feedbackElement.hidden = false;
+      feedbackElement.style.color = 'red';
+    }
   };
 
-  const onSignOutHandler = (e) => {
+  const onUpdatePostHandler = async (e) => {
     e.preventDefault();
-    signOut();
+    const { parentElement } = e.target;
+    const message = parentElement.message.value;
+    const updatedAt = new Date();
+    const id = parentElement.dataset.id;
+
+    await firebaseClient.post.update(id, { message, updatedAt });
+    const posts = state.posts.map(
+      (post) => (post.id === id ? { ...post, message, updatedAt } : post),
+    );
+    setData({ posts });
+    showPosts();
+  };
+
+  const onRemovePostHandler = (e) => {
+    const { parentElement } = e.target;
+    const { id } = parentElement.dataset;
+    firebaseClient.post.remove(id);
+    const posts = state.posts.filter((post) => post.id !== id);
+    setData({ posts });
+    showPosts();
+  };
+
+  const onEditPostHandler = (e) => {
+    e.target.hidden = true;
+    const { parentElement } = e.target;
+    const selectedPost = findPostById(parentElement.dataset.id);
+    parentElement.innerHTML = EditablePost(selectedPost);
+    parentElement.firstElementChild.firstElementChild.focus();
+  };
+
+  const onEnableUpdateButtonHandler = (e) => {
+    const { parentElement, value: message } = e.target;
+    const { message: currentMessage } = findPostById(parentElement.dataset.id);
+
+    const buttonElement = parentElement.querySelector('button[data-behavior=onUpdatePost]');
+    buttonElement.disabled = message === currentMessage;
   };
 
   const render = (target) => {
     target.innerHTML = html();
 
-    document.getElementById('submitBtn').addEventListener('click', onCreatePostHandler);
+    document.getElementById('postBuilderForm').addEventListener('submit', onCreatePostHandler);
     document.getElementById('signOut').addEventListener('click', onSignOutHandler);
+    document.addEventListener('click', (e) => {
+      const { behavior } = e.target.dataset;
+      if (behavior === 'onEditPost') {
+        onEditPostHandler(e);
+      }
+
+      if (behavior === 'onCancel') {
+        showPosts();
+      }
+
+      if (behavior === 'onRemovePost') {
+        onRemovePostHandler(e);
+      }
+
+      if (behavior === 'onUpdatePost') {
+        onUpdatePostHandler(e);
+      }
+    });
+
+    document.addEventListener('keyup', (e) => {
+      const { behavior } = e.target.dataset;
+
+      if (behavior === 'onEnableUpdateButton') {
+        if (e.key === 'Escape') {
+          showPosts();
+        } else {
+          onEnableUpdateButtonHandler(e);
+        }
+      }
+    });
```

Comment:
Use the `input` [event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event) to fire `onEnableUpdateButtonHandler`